### PR TITLE
Improve definition of "disabled element"

### PR DESCRIPTION
**Pull request by Jym77**
The definition was mimicking https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#concept-fe-disabled rather than the more generic https://html.spec.whatwg.org/multipage/semantics-other.html#concept-element-disabled 
As far as I follow, this resulted in `option` elements never being disabled in our sense (well or indirectly via the `aria-disabled` attribute) and therefore, labels of `<option disabled>` where applicable for the colour contrast rule.

This is changing the def to the more generic one, using the CSS selector directly (so it also applies to non-HTML elements).
Additionally, the assumption in that definition is copied in the rule with indication of the consequences if it does not hold.

(noticed the issue while working on #1455)

Closes issue(s):

- N/A

Need for Final Call:
This will require a 1 week Final Call (indirectly changing Applicability of a rule)

---

### **When creating PR:**

- [X] Make sure you're requesting to **pull a branch** (right side) to the `develop` branch (left side).
- [X] Make sure you do not remove the "How to Review and Approve" section in your pull request description

### **After creating PR:**

- [X] Add yourself (and co-authors) as "Assignees" for PR.
- [X] Add label to indicate if it's a `Rule`, `Definition` or `Chore`.
- [ ] [Link the PR to any issue it solves](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue). This will be done automatically by [referencing the issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) at the top of this comment in the indicated place.
- [X] Optionally request feedback from anyone in particular by assigning them as "Reviewers".

### **When merging a PR:**

- [ ] Close any issue that the PR resolves. This will happen automatically upon merging if the PR was correctly linked to the issue, e.g. by referencing the issue at the top of this comment.

## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Final Call period. In case of disagreement, the longer period wins.

### CODE REVIEW

File: pages/glossary/disabled-element.md [hunk 1]
```diff
@@ -7,38 +7,17 @@ input_aspects:
   - DOM tree
 ---
 
-An element is considered disabled when it has been rendered [inoperable][] using one of the following properties:
+An element is _disabled_ when it has been rendered [inoperable][] in one of the following ways:
 
-1. The [`disabled`][disabled] attribute. The presence of this attribute, regardless of its value, on a
+1. The element matches the [`:disabled` pseudo-class][disabled pseudo-class], for HTML elements, it means that the element is [actually disabled][].
```
Comment (kasperisager) [inline pages/glossary/disabled-element.md:12]:
```suggestion
1. The element matches the [`:disabled` pseudo-class][disabled pseudo-class]. For HTML elements this means that the element is [actually disabled][].
```

**Review by kasperisager — APPROVED** [review]
Just a small editorial suggestion, looks good otherwise!

File: _rules/text-contrast-afw4f7.md [hunk 2]
```diff
@@ -46,6 +46,8 @@ For each test target, the [highest possible contrast][] between the [foreground
 
 - Text that has the same foreground and background color (a contrast ratio of 1:1) is not considered to be "visual presentation of text", making it inapplicable to the success criterion. Text hidden in this way can still cause accessibility issues under other success criteria, depending on the content.
 
+- The definition of [disabled element][disabled] assumes that the presence of the `aria-disabled` attribute is matched by similar techniques for users not relying on assistive technologies. If this is not the case, that definition may produce incorrect results and in consequence this rule might be Inapplicable to some text nodes that still require a good contrast ratio.
```
Comment (WilcoFiers) [inline _rules/text-contrast-afw4f7.md:49]: This phrase "matched by similar techniques for users not relying on assistive technologies" is a little vague. Can you clarify it?

File: pages/glossary/disabled-element.md [hunk 3]
```diff
@@ -7,38 +7,17 @@ input_aspects:
...
-   - [`button`][button],
-   - [`input`][input],
-   - [`select`][select],
-   - [`textarea`][textarea],
-   - [form-associated custom element][],
-   - or [`fieldset`][fieldset] element
+2. The element has a [shadow-including ancestor][] whose `aria-disabled` [attribute value][] is "true".
 
-   will disable the element itself if it is not a [`fieldset`][fieldset] or, if it is, will disable any [descendants][descendant] of the element, excluding those that are [descendants][descendant] of the first [`legend`][legend] [child][] of the element.
+   **Note:** When the `aria-disabled` attribute is specified on an element, it is assumed that the element has also been disabled for users that do not rely on [assistive technology][]. For example, this can be done by disabling pointer events using the `pointer-events` property and by disabling keyboard interactions using the `tabindex` attribute.
```
Comment (WilcoFiers) [inline pages/glossary/disabled-element.md:16]: This should probably go under an assumptions heading instead.

**Review by WilcoFiers — CHANGES_REQUESTED** [review]

File: _rules/text-contrast-afw4f7.md (hunk 2 above)
Comment (Jym77) [reply _rules/text-contrast-afw4f7.md]: Bad copy/paste from the def 😞 Corrected.

**Review by WilcoFiers — APPROVED** [review]

**Review by ShadowBB — APPROVED** [review]

**Comment by Jym77** [comment]
Final Call ends on November 24.

**Comment by Jym77** [comment]
Final Call has ended. Merging.

**Pull request merged**